pypeclub/OpenPype
# General: Move create logic from avalon to OpenPype

## Brief description
Move "Create workflow" from avalon-core into OpenPype.

## Description
Create workflow in avalon consist of 3 parts: `Creator` plugin, `create` function and `CreatorError`.
- Exception `CreatorError` is already available in openpype.
- `Creator` is renamed to `LegacyCreator`
- `create` function is renamed to `legacy_create`

## Changes
- added `LegacyCreator` and `legacy_create` into `openpype/pipeline/create/legacy_create.py`
- removed `PypeCreatorMixin` and `Creator` from `openpype/plugin.py` their content was added to `LegacyCreator`
- changed imports to use `LegacyCreator`, `legacy_creator` and `CreatorError` from `openpype.pipeline`
- removed `Creator` from Photoshop which was not used and had invalid code
- patched discover also checks discovered plugin and check if found plugins are subclasses of passed superclass

## Testing notes:
1. Run host
2. Show creator tool
3. Check if creators are available
4. Try to trigger a creation (at least one Creator)

### Tested hosts
- [x] Standalone Publisher

Depends on https://github.com/pypeclub/avalon-core/pull/427
